File: sorting/c/selection_sort.c
#include "utils.h"
#include <stdio.h>

void selection_sort(int arr[], int n) {
  for (int i = 0; i < n - 1; ++i) {
    int minIndex = i;

    for (int j = i + 1; j < n; ++j) {
      if (arr[j] < arr[minIndex]) {
        minIndex = j;
      }
    }

    if (minIndex != i) {
      swap(&arr[i], &arr[minIndex]);
    }
  }
}

int main() {
  int arr[] = {98, 2, -10, 9, 0, 7};
  int n = sizeof(arr) / sizeof(arr[0]);

  printf("Original array: ");
  printArray(arr, n);
  selection_sort(arr, n);

  printf("Selection sorted array: ");
  printArray(arr, n);
  return 0;
}

File: sorting/javascript/selection_sort.js
function selectionSort(arr) {
  let len = arr.length;

  for (let i = 0; i < len - 1; i++) {
    let minIndex = i;

    // Find the index of the minimum element in the remaining unsorted part
    for (let j = i + 1; j < len; j++) {
      if (arr[j] < arr[minIndex]) {
        minIndex = j;
      }
    }

    // Swap the found minimum element with the first element
    if (minIndex !== i) {
      let temp = arr[i];
      arr[i] = arr[minIndex];
      arr[minIndex] = temp;
    }
  }

  return arr;
}

let unsortedArray = [64, 25, 12, 22, 11];
let sortedArray = selectionSort(unsortedArray);
console.log("Sorted Array:", sortedArray);
